_posts/2024-08-03-gsoc2024_woods.markdown

```diff
@@ -0,0 +1,261 @@
+
+u = mda.Universe(COORDINATES_TOPOLOGY, "s3://sample-bucket-name/COORDINATES_SYNTHETIC_H5MD.h5md")
+for ts in u.trajectory:
+    pass
+```
+
+You can follow this same process for reading `.zarrmd` files with the added advantage
+that Zarrtarj can write `.zarrmd` files directly into an S3 bucket.
+
+### Writing trajectories from MDAnalysis into a zarrmd file in an S3 Bucket 
+
+Using the same credentials with read/write access, you can write a trajectory
+into your bucket.
+
+You can change the stored precision of floating point values in the file with the optional
+`precision` kwarg and pass in any `numcodecs.Codec` compressor with the optional
+`compressor` kwarg. See [numcodecs](https://numcodecs.readthedocs.io/en/stable/)
+for more on the available compressors.
+
+Chunking is automatically determined for all datasets to be optimized for
+cloud storage and is not configurable by the user. 
+Initial benchmarks show this chunking strategy is effective for disk storage as well.
+
+```python
+import zarrtraj
+import MDAnalysis as mda
+from MDAnalysisTests.datafiles import PSF, DCD
+import numcodecs
+import os
+
+os.environ["AWS_PROFILE"] = "sample_profile"
+os.environ["AWS_REGION"] = "us-west-1"
+
+u = mda.Universe(PSF, DCD)
+with mda.Writer("s3://sample-bucket-name/test.zarrmd", 
+                n_atoms=u.trajectory.n_atoms, 
+                precision=3,
+                compressor=numcodecs.Blosc(cname="zstd", clevel=9)) as W:
+                for ts in u.trajectory:
+                    W.write(u.atoms)
+
+```
+
+If you have additional questions, please don't hesitate to open a discussion on the [zarrtarj github](https://github.com/Becksteinlab/zarrtraj).
+The [MDAnalysis discord](https://discord.com/channels/807348386012987462/) is also a 
+great resource for asking questions and getting involved in MDAnalysis.
+
+## What's next for Zarrtraj?
+
+Zarrtraj is currently in a fully operational state and is ready for use!
+However, I'm excited about creating some new features in the future that will
+make Zarrtraj more flexible and faster.
+
+### Lazytimeseries
+
+In MDAnalysis, many trajectory readers expose a `timeseries` method for getting access to
+coordinate data for a subselection of atoms across a trajectory. This provides 
+a viable way to sidestep the `Timestep` (eagerly-loaded frame-based) paradigm that 
+MDAnalysis uses for handling trajectory data. Zarrtraj could implement a 
+"lazytimeseries" that returns a lazy dask array of a selection of atoms' positions
+across the trajectory. Early benchmarks show that analysis based on such a lazy array
+can outperform `Timestep`-based analysis.
+
+### Asynchronous Reading
+
+The performance impact of network IO could be reduced by creating a multi-threaded `ZARRH5MDReader` that isn't blocked
+by analysis code executing. The reader could eagerly load the cache with the next frames
+the analysis code will need to reduce the impact of network IO on exeution time.
+
+## Acknowledgements
```

Comment:
It might be a good idea to thank Google here as well :)

Reply:
Just added it!